packages/medusa/src/services/event-bus.ts

```diff
@@ -1,143 +1,61 @@
-import Bull from "bull"
-import Redis from "ioredis"
-import { isDefined } from "medusa-core-utils"
 import { EntityManager } from "typeorm"
-import { ulid } from "ulid"
+import {
+  AbstractEventBusModuleService,
+  Subscriber,
+  SubscriberContext,
+  TransactionBaseService,
+} from "../interfaces"
 import { StagedJob } from "../models"
-import { StagedJobRepository } from "../repositories/staged-job"
-import { ConfigModule, Logger } from "../types/global"
 import { sleep } from "../utils/sleep"
-import JobSchedulerService from "./job-scheduler"
+import StagedJobService from "./staged-job"
 
 type InjectedDependencies = {
   manager: EntityManager
-  logger: Logger
-  stagedJobRepository: typeof StagedJobRepository
-  jobSchedulerService: JobSchedulerService
-  redisClient: Redis.Redis
-  redisSubscriber: Redis.Redis
+  stagedJobService: StagedJobService
+  eventBusModuleService: AbstractEventBusModuleService
 }
 
-type Subscriber<T = unknown> = (data: T, eventName: string) => Promise<void>
+export default class EventBusService extends TransactionBaseService {
```

Comment:
self-review: The core `EventBusService` has been changed to act as a relayer of messages to the installed event bus service